src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,97 @@
+import Issue from "../../issueHandling/issue";
+import { concatOneIssueList } from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import { UnexpectedError } from "./Errors";
+import modes from "./predicateList.json";
+
+export type Mode = "propositionalLogic" | "firstOrderPredicateLogic" | "test";
+
+/**
```

Comment:
I think you are describing the function too much in detail and trying to justify it.
The JSDoc should:

- Describe what the function achieves
- Document what the functions parameters should be with @param
- Document what the return will be (if it is not null) with @return
- Document which Exceptions it might throw and when (if any) with @throws (this isn't usually used a lot it seems, but it seems useful to me)

This is very useful because most IDEs will tell you the information that you have entered when you using the function somewhere else. It can also automatically be compiled to an actual documentation and give documentation a more structured character.

Justifications for this functions should usually be in non-JSDoc comments.

Here is an example
https://jsdoc.app/howto-es2015-classes.html

And here some more details on how to write a few things:
https://devhints.io/jsdoc

Reply:
You're right. This justification would be better fitted in some kind of design document.

Reply:
Since we're writing Typescript (instead of Javascript), I think we should omit the type-info in JSDoc-comments though. This can be expressed better with the actual parameter-types / method-return-type.